test/Microsoft.AspNet.Mvc.ModelBinding.Test/Binders/KeyValuePairModelBinderTest.cs

```diff
@@ -28,8 +26,7 @@ public class KeyValuePairModelBinderTest
             var retVal = await binder.BindModelAsync(bindingContext);
 
             // Assert
-            Assert.NotNull(retVal);
-            Assert.Null(retVal.Model);
+            Assert.Null(retVal);
```

Comment:
> the only non-comment change I made to KeyValuePairModelBinder was to remove extra special case returns, leaving that to the existing handling in GenericModelBinder

If it's not visible why not just return a value then? I don't get why you choose to do it differently when it doesn't affect the outcome of the system. It's much more conceptually symmetric to return a value - that indicates that the KVPMB did *indeed* do something. Returning `null` just requires extra explanation even if it does result in the right outcome.